src/module.ts

```diff
@@ -38,23 +38,21 @@ export default defineNuxtModule<ModuleOptions>({
   setup(options, nuxt) {
     const logger = useLogger('turnstile')
     const siteKey = options.siteKey || nuxt.options.runtimeConfig.public?.turnstile?.siteKey
+
     if (!siteKey) {
       logger.warn(
         'No site key was provided. Make sure you pass one at runtime by setting NUXT_PUBLIC_TURNSTILE_SITE_KEY.'
       )
     }
 
-    if (options.secretKeyPath) {
-      try {
-        options.secretKey = fs.readFileSync(
-          resolve(nuxt.options.rootDir, options.secretKeyPath),
-          'utf-8'
-        )
-      } catch {}
+    const secretKey = options.secretKeyPath
+      ? fs.readFileSync(resolve(nuxt.options.rootDir, options.secretKeyPath), 'utf-8')
+      : options.secretKey || nuxt.options.runtimeConfig.turnstile?.secretKey
 
-      if (!options.secretKey) {
-        logger.warn(`No secret key present in \`${options.secretKeyPath}\`.`)
-      }
+    if (!secretKey) {
+      logger.warn(
+        'No secret key was provided. Make sure you pass one at runtime by setting NUXT_TURNSTILE_SECRET_KEY.'
```

Comment:
if a secret key is provided in `.env` this warning will wrongly trigger because it won't be present in `nuxt.options.runtimeConfig`. We have new utilities in kit from v3.12 that will make this easier in future: https://github.com/nuxt/nuxt/pull/27117.

But for now if we want this warning we should move the warning to the runtime code.

Reply:
I'll wait for Nuxt 3.12 which looks to be landing soon.